Docs: added documentation for the "prepare time series"-transformation. (#36761)
grafana/grafana: docs/sources/panels/transformations/types-options.md

- [Rows to fields]({{< relref "./rows-to-fields" >}})
- [Series to rows]({{< relref "./types-options.md#series-to-rows" >}})
- [Sort by]({{< relref "./types-options.md#sort-by" >}})
- [Prepare-time-series]({{< relref "./types-options.md#prepare-time-series" >}})

Keep reading for detailed descriptions of each type of transformation and the options available for each, as well as suggestions on how to use them.

## Reduce

> The _Reduce_ transformation will apply a calculation to each field in the frame and return a single value. Time fields are removed when applying

Reviewer: Change from passive to active voice:

> This transformation removes any time fields when applied.

> The _Reduce_ transformation will apply a calculation to each field in the frame and return a single value. Time fields are removed when applying

Reviewer: ```suggestion
The _Reduce_ transformation applies a calculation to each field in the frame and return a single value. Time fields are removed when applying
```

this transformation.

Consider the input:

Query A:

| Time                | Temp | Uptime  |
| ------------------- | ---- | ------- |
| 2020-07-07 11:34:20 | 12.3 | 256122  |
| 2020-07-07 11:24:20 | 15.4 | 1230233 |

Query B:

| Time                | AQI | Errors |
| ------------------- | --- | ------ |
| 2020-07-07 11:34:20 | 6.5 | 15     |
| 2020-07-07 11:24:20 | 3.2 | 5      |

The reduce transformer has two modes:

- **Series to rows -** Creates a row for each field and a column for each calculation.
- **Reduce fields -** Keeps the existing frame structure, but collapses each field into a single value.

For example, if you used the **First** and **Last** calculation with a **Series to rows** transformation, then
the result would be:

| Field  | First  | Last    |
| ------ | ------ | ------- |
| Temp   | 12.3   | 15.4    |
| Uptime | 256122 | 1230233 |
| AQI    | 6.5    | 3.2     |
| Errors | 15     | 5       |

The **Reduce fields** with the **Last** calculation,
results in two frames, each with one row:

Query A:

| Temp | Uptime  |
| ---- | ------- |
| 15.4 | 1230233 |

Query B:

| AQI | Errors |
| --- | ------ |

> | 3.2 | 5      |

Reviewer 2: not related? :)

Author: Yeah, it is prettier that formats the document. Maybe I should revert those?

With the transformation applied, you can see we are left with just the remainder of the string.

{{< figure src="/static/img/docs/transformations/rename-by-regex-after-7-3.png" class="docs-image--no-shadow" max-width= "1100px" >}}

## Prepare time series

> **Note:** This transformation is available in Grafana 7.5.10+ and Grafana 8.0.6+.

> You might end up in a situation where a data source returns time series data in a time series format that the panel you want to use does not support ([read more about the different data frame formats here]({{< relref "../../developers/plugins/data-frames.md" >}})).

Reviewer 3: ```suggestion
You might end up in a situation where a data source returns time series data in a format that isn't supported by the panel you want to use. [Read more about the different data frame formats here]({{< relref "../../developers/plugins/data-frames.md" >}}).
```

> You might end up in a situation where a data source returns time series data in a time series format that the panel you want to use does not support ([read more about the different data frame formats here]({{< relref "../../developers/plugins/data-frames.md" >}})).

Reviewer 2: ```suggestion
Prepare time series transformation is useful when a data source returns time series data in a format that the panel you want to use does not support ([read more about the different data frame formats here]({{< relref "../../developers/plugins/data-frames.md" >}})).
```

> This transformation will help you resolve this issue by converting the time series data from either the wide format to the long format or the other way around.

Reviewer: ```suggestion
This transformation helps you resolve this issue by converting the time series data from either the wide format to the long format or the other way around.
```
